### DISCUSSION TITLE
ecFSEOF implementation in GECKO 3.1.2

### DISCUSSION THREAD

**Discussion by JBiaggi** [SysBioChalmers/GECKO#360] [GECKO3]
Hello,

I was checking the new ecFSEOF function that you published in the last release of GECKO and I noticed one likely miscalculation of the slopes in lines 149 and 157 of you code:

`slope_rxns(i)  = abs(v_matrix(i,nSteps-1)-v_matrix(i,1))/abs(maxTarget-maxTarget/nSteps-1);
`

In the maxTarget part, it seems to be subtracting 1 from maxTarget-maxTarget/nSteps. Is this intentional?

In my forked repo I put nSteps-1 between parenthesis and got results that are more similar to the output of the FSEOF function of RAVEN.

**Comment by edkerk** [comment]
@ae-tafur, any idea?

**Comment by ae-tafur** [comment]
Hej, @JBiaggi  thanks for checking. What you report is true, any removing the `-1 ` or putting `nSteps-1` between parenthesis will get similar results as RAVEN implementation for `rxnTargets`.

That was intentional in order to double slope value because in ecModels the incorporation of enzyme constraints fluxes values go down. Then, calculating the slope will result also in small numbers which can be filtered in ecFSEOF, specially for no growth coupled metabolites. I tested FSEOF for acetic acid as target, both, RAVEN, GECKO v3.1.2 and your suggestion. Results shows that GECKO implementation is able to predict more targets (including those predicted in RAVEN) but  removing the `-1 ` or putting `nSteps-1` between parenthesis in ecFSEOF does not have any change more than slope value. 

 Moreover, notice despite ecFESOF report a rxnList ( `rxnTargets`) this implementation also identify individual gene targets `geneTargets` where lower slope values could have a big impact.

please see the results here:

[validation_ecFSEOF_RAVEN.txt](https://github.com/SysBioChalmers/GECKO/files/13565079/validation_ecFSEOF_RAVEN.txt)
[validation_ecFSEOF_v3.txt](https://github.com/SysBioChalmers/GECKO/files/13565077/validation_ecFSEOF_v3.txt)
[validation_ecFSEOF_bug_sol.txt](https://github.com/SysBioChalmers/GECKO/files/13565076/validation_ecFSEOF_bug_sol.txt)

**Comment by JBiaggi** [comment]
Understood @ae-tafur. Thanks for explaining the rationale behind the new release!
